test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -896,6 +1021,81 @@ UCS_TEST_P(test_ucp_sockaddr, err_handle_without_err_cb)
     EXPECT_EQ(1u, sender().get_err_num());
 }
 
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr, force_close_during_rndv,
+                     (send_recv_type() != SEND_RECV_TAG), "RNDV_THRESH=0")
+{
+    constexpr size_t length = 4 * UCS_KBYTE;
+
+    listen_and_communicate(false, SEND_DIRECTION_BIDI);
+
+    mem_buffer recv_buffer(length, UCS_MEMORY_TYPE_HOST);
+    recv_buffer.pattern_fill(2, length);
+    void *rreq = recv(receiver(), recv_buffer.ptr(), length,
+                      rtag_complete_check_data_cbx,
+                      reinterpret_cast<void*>(&recv_buffer));
+
+    mem_buffer send_buffer(length, UCS_MEMORY_TYPE_HOST);
+    send_buffer.pattern_fill(1, length);
+    void *sreq = send(sender(), send_buffer.ptr(), length,
+                      send_recv_type(), scomplete_reset_data_cbx,
+                      reinterpret_cast<void*>(&send_buffer));
+
+    ucp_ep_h ep         = sender().revoke_ep();
+    ucs_time_t deadline = ucs::get_deadline();
+    do {
+        ucp_worker_progress(sender().worker());
+    } while (ucs_hlist_is_empty(&ucp_ep_ext_gen(ep)->proto_reqs) &&
+             (ucs_get_time() < deadline));
+    EXPECT_FALSE(ucs_hlist_is_empty(&ucp_ep_ext_gen(ep)->proto_reqs));
+
+    using funcs_t = std::pair<uct_ep_flush_func_t, uct_ep_pending_add_func_t>;
```

Comment:
maybe the value could be uct_iface_ops_t?

Reply:
> maybe the value could be uct_iface_ops_t?

then we should save all operations which is much slower than save only two

Reply:
code would be simpler

Reply:
done